src/GraphQL.Tests/Execution/ResolveFieldContextTests.cs

```diff
@@ -42,14 +42,15 @@ public void argument_converts_long_to_int()
         public void long_to_int_should_throw_for_out_of_range()
         {
             long val = 89429901947254093;
-            _context.Arguments["a"] = val;
+            _context.Arguments["a"] = new ArgumentValue(val, ArgumentSource.Literal);
```

Comment:
```c#
public readonly struct ArgumentValue
{
   public static implicit operator ArgumentValue(string value) => new ArgumentValue(value, ArgumentSource.Literal);
   public static implicit operator ArgumentValue(int value) => new ArgumentValue(value, ArgumentSource.Literal);
...
}
```

At least fewer changes will be required in tests. Although there is some convention in the fact that the value of `Source` is fixed as literal.

Reply:
And `_context.Arguments["a"] = val;` just works again.

Reply:
```c#
   var d = new Dictionary<string, ArgumentValue>
            {
                ["ddd"] = "adfafaf",
                ["adfaf"] = 5,
                ["afafaf"] = new object[] { true, 7, "f" }
            };
```

Reply:
I feel like this leaves open an easy possibility that someone can set an `ArgumentValue` as a literal without realizing it.  Of course it simplifies the tests, but it hurts the codebase.

Reply:
> I feel like this leaves open an easy possibility that someone can set an ArgumentValue as a literal without realizing it.

This can be said for any implicit cast operator. Nevertheless, they are used quite widely. If there is no loss of accuracy, then why not? In this case, there is no loss of precision, we just chose the default behavior. Moreover, consumers are very unlikely to **create** this structs. They will **consume** them already populated by us.

Reply:
You cannot write an implicit operator for `object`. There are an infinite number of types, so now we need to select which types we are writing implicit operators for.  This does not help the codebase and only helps tests.  Moreover, all the required code changes for the tests have already been completed.  If it had not been, that's one thing.  But they are done.  To be of any value, they would all need to change back.  And finally, it is better for tests to explicitly define the behavior they are testing.

If we want a helper to set argument values, let's write it within the test project.  We could write an extension method that converts `IDictionary<string, object>` to `Dictionary<string, ArgumentValue>`.  Then code might look like this:

```cs
   var d = new Dictionary<string, object>
            {
                ["ddd"] = "adfafaf",
                ["adfaf"] = 5,
                ["afafaf"] = new object[] { true, 7, "f" }
            }.ToArgumentValues();
```

Either way, if you want to change it, go ahead.  I don't see any reason to spend time on it.

Reply:
ok